spec/predicates/runtime-trace.md

```diff
@@ -0,0 +1,2677 @@
+
+## Schema
+
+The schema of the attestation type is the core part of the document. It defines the fields included in an instantiation of the attestation type, as well as rules to parse them.
+
+```json
+{
+  "_type": "https://in-toto.io/Statement/v0.1",
+  "predicateType": "https://in-toto.io/runtime-trace/v0.1.0",
+  "subject": [{ ... }],
+  "predicate": {
+    "monitor": {
+      "id": "<URI>"
+    },
+    “build”: {
+        “builderId”: “<URI>”,
+        “type”: “<URI>”,
+        “event”: “<STRING>”,
+    },
+    "invocation": {
+     "configSource": {
+        "uri": "<URI>",
+        "digest": { /* DigestSet */ },
+        "entryPoint": "<STRING>"
+      },
+      "tracePolicy": {}
+    },
+    "monitorLog": {
+      "process": [
+         { /* object */ }
+      ],
+      "fileAccess": [
+        { /* object */ }
+      ],
+    },
+    "metadata": {
+      "buildStartedOn": "<TIMESTAMP>",
+      "buildFinishedOn": "<TIMESTAMP>",
+    }
+  }
+}
+```
+
+### Parsing Rules
+
+This predicate follows the in-toto attestation parsing rules. Summary:
+
+-   Consumers MUST ignore unrecognized fields.
+-   The `predicateType` URI includes the major version number and will always change whenever there is a backwards incompatible change.
+-   Minor version changes are always backwards compatible and “monotonic.” Such changes do not update the `predicateType`.
+-   Producers MAY add extension fields using field names that are URIs.
+-   _optional_ fields MAY be unset or null, and should be treated equivalently. Both are equivalent to empty for object or array values.
+
+## Fields
+
+`monitor` _object_, _required_
+
+Identifies the specific monitor instance used to trace the runtime.
+
+`monitor.id` _string (URI)_, _required_
+
+URI indicating the monitor’s identity.
+
+`build` _object_, _required_
+
+Feedback required: Should this predicate be scoped to runtime traces of builds only or generalize to runtime traces of any activity?
+
+`builder.builderId` _string (URI)_, _required_
+
+URI indicating the builder’s identity. Ex: GitLab Runner ID
+
+`build.type` _string (TypeURI)_, _required_
+
+URI indicating the type of build performed.
+
+`build.event` _string_, _required_
+
+String identifying the specific job or task associated with the attestation.
+
+`invocation` _object_, _optional_
+
+Identifies the event that initiated the monitor. Similar to `invocation` in SLSA Provenance.
+
+`invocation.configSource` _object_, _optional_
+
+Effectively a pointer to the monitor's configuration.
+
+`invocation.tracePolicy` _object_, _optional_
+
+Feedback required: Should tracingPolicy point outwards to a separate attestation with the same subject that wraps around the actual policy? The policy can be more securely defined and distributed (signed for by developer keys rather than online keys for example).
+
+`monitorLog` _object_, _required_
```

Comment:
probably very subjective but trace might be more descriptive in this case than monitor. the attestation is also called runtime *trace*. and it's definition is a *trace* policy. so calling this monitor here is no standing out for me

Reply:
Fair! I think we've got a couple of different field rename suggestions to consider and implement here.